packages/tinacms/src/admin/pages/CollectionListPage.tsx

```diff
@@ -1298,27 +1302,61 @@ interface FolderModalProps {
   createFunc(): void;
   folderName: string;
   setFolderName(folderName: string): void;
+  validationRegex?: string;
 }
 
 const FolderModal = ({
   close,
   createFunc,
   folderName,
   setFolderName,
+  validationRegex,
 }: FolderModalProps) => {
+  const [isFolderNameValid, setIsFolderNameValid] = useState(false);
+  const [isInteracted, setIsInteracted] = useState(false);
+
+  useEffect(() => {
+    validateFolderName(folderName);
+  }, [folderName]);
+
+  const validateFolderName = (name: string) => {
+    if (!validationRegex || !name.trim()) {
+      setIsFolderNameValid(!!name.trim());
+      return !!name.trim();
+    }
+
+    try {
+      const regex = new RegExp(validationRegex);
+      const valid = regex.test(name);
+      setIsFolderNameValid(valid);
+      return valid;
+    } catch (error) {
+      setIsFolderNameValid(false);
+      return false;
+    }
+  };
+
   return (
     <Modal>
       <PopupModal>
         <ModalHeader close={close}>Create Folder</ModalHeader>
         <ModalBody padded={true}>
           <>
-            {/* <p className='mb-4'>
-            </p> */}
             <BaseTextField
               placeholder='Enter the name of the new folder'
               value={folderName}
-              onChange={(event) => setFolderName(event.target.value)}
+              className={`mb-4 ${!isFolderNameValid ? 'border-red-500' : ''}`}
```

Comment:
```suggestion
              className={`mb-4 ${!isFolderNameValid && isInteracted ? 'border-red-500' : ''}`}
```

Small change for consistent behavior